xbmc/pvr/timers/PVRTimers.cpp

```diff
@@ -248,6 +248,47 @@ bool CPVRTimers::UpdateEntries(const CPVRTimers &timers)
     UpdateEpgEvent(*timerIt);
   }
 
+  /* go through the timer list and check for parent timers */
+  for (MapTags::const_iterator it = timers.m_tags.begin(); it != timers.m_tags.end(); ++it)
+  {
+    for (VecTimerInfoTag::const_iterator timerIt = it->second->begin(); timerIt != it->second->end(); ++timerIt)
+    {
+      /* check if the parent timer is present */
+      CPVRTimerInfoTagPtr parentTimer = GetByClient((*timerIt)->m_iClientId, (*timerIt)->m_iParentClientIndex);
+      if (parentTimer)
+      {
+        CPVRTimerInfoTagPtr childTimer = GetByClient((*timerIt)->m_iClientId, (*timerIt)->m_iClientIndex);
+        if (childTimer)
+        {
+          switch (childTimer->m_state)
+          {
+          case PVR_TIMER_STATE_NEW:
+          case PVR_TIMER_STATE_SCHEDULED:
+          case PVR_TIMER_STATE_CONFLICT_OK:
+            parentTimer->m_iActiveChildTimers++;
```

Comment:
I would do this very different, btw. Add a new method <code>bool CPVRTimerInfoTag::UpdateChildStatus(const CPVRTimerInfoTagPtr &child)</code> and move the code that currently sets the new members to it. This way, you will not need the setters for the members as they will not be accessed from outside the class. Anyway, make the new members private, please. 

Then, here would stay just:

<pre>
if (childTimer)
 parentTimer->UpdateChildState(childTimer);
</pre>